### PR TITLE
Always specify coffeelint.json as default config

### PR DESCRIPTION
**Pull request by ABaldwinHunter**
This sidesteps an issue with resolving the config and location of node_modules
for coffeelint projects using plugins:
https://github.com/clutchski/coffeelint/issues/571

@codeclimate/review

### CODE REVIEW

**Comment by gdiggs** [comment]
Does this work if there is no `coffeelint.json` present?

**Comment by ABaldwinHunter** [comment]
@GordonDiggs ah no it does not. Good catch, thanks.

**Comment by ABaldwinHunter** [comment]
@GordonDiggs updated!

File: lib/cc/engine/coffeelint_results.rb [hunk 1]
```diff
@@ -3,23 +3,35 @@
...
           JSON.parse(`#{cmd}`)
         end
       end
+
+      private
+
+      def config_path
+        if @config
+          @config
+        elsif File.exist? DEFAULT_CONFIG_PATH
```
Comment (gdiggs) [inline lib/cc/engine/coffeelint_results.rb:33]: 💄 parentheses around the arguments here

File: lib/cc/engine/coffeelint_results.rb [hunk 2]
```diff
@@ -3,23 +3,35 @@
...
         Dir.chdir(@directory) do
           JSON.parse(`#{cmd}`)
         end
       end
+
+      private
+
+      def config_path
+        if @config
+          @config
```
Comment (gdiggs) [inline lib/cc/engine/coffeelint_results.rb]: Can you turn this into an `attr_reader`?

File: lib/cc/engine/coffeelint_results.rb [hunk 3]
```diff
@@ -3,23 +3,37 @@
 module CC
   module Engine
     class CoffeelintResults
+      DEFAULT_CONFIG_PATH = "/code/coffeelint.json".freeze
+
       def initialize(directory, files, config)
         @directory = directory
-        @config = config
         @files = files
+        @config = config
```
Comment (ABaldwinHunter) [inline lib/cc/engine/coffeelint_results.rb:11]: reordered for style

**Comment by ABaldwinHunter** [comment]
@GordonDiggs ready for another look!

File: lib/cc/engine/coffeelint_results.rb [hunk 4]
```diff
@@ -3,23 +3,37 @@
...
+        cmd << " -f #{config_path}" if config_path
         cmd << " -q --reporter raw #{escaped_files}"
         Dir.chdir(@directory) do
           JSON.parse(`#{cmd}`)
         end
       end
+
+      private
+
+      attr_reader :config
```
Comment (gdiggs) [inline lib/cc/engine/coffeelint_results.rb]: Sorry if I wasn't clear before - I think this is the opportunity for all 3 ivars to become `attr_reader`s and accessed that way. It seems worse to only have one

**Comment by ABaldwinHunter** [comment]
@GordonDiggs good call on attr_readers. Updated

**Comment by gdiggs** [comment]
LGTM

**Pull request merged**